product_seasonality/models/seasonal_config_line.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2021 Camptocamp SA
+# @author: Julien Coux <julien.coux@camptocamp.com>
+# @author: Simone Orsi <simone.orsi@camptocamp.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class SeasonalConfLine(models.Model):
+    _name = "seasonal.config.line"
+    _description = "Seasonal configuiration Line"
+
+    seasonal_config_id = fields.Many2one(comodel_name="seasonal.config", required=True)
+    product_id = fields.Many2one(
+        comodel_name="product.product",
+        domain=[("sale_ok", "=", True)],
+        required=True,
+    )
+    date_start = fields.Datetime(required=True)
+    date_end = fields.Datetime()
+    monday = fields.Boolean(default=True)
+    tuesday = fields.Boolean(default=True)
+    wednesday = fields.Boolean(default=True)
+    thursday = fields.Boolean(default=True)
+    friday = fields.Boolean(default=True)
+    saturday = fields.Boolean(default=True)
+    sunday = fields.Boolean(default=True)
+
+    @api.constrains("date_start", "date_end")
+    def _check_date_end(self):
+        for line in self:
+            if line.date_end and line.date_end < line.date_start:
+                raise ValidationError(
+                    _("The end date cannot be earlier than the start date.")
+                )
+
+    def is_sale_ok(self, date):
+        # TODO: can't we just get a date and determin week day from it?
```

Comment:
With the current way it's easy to check if the right boolean is checked or not: `self["monday"]` by example will directly give you the value of the flag.